client/app/pages/dashboards/dashboard.js

```diff
@@ -392,7 +392,12 @@ function DashboardCtrl(
   }
 
   this.openShareForm = () => {
-    ShareDashboardDialog.showModal({ dashboard: this.dashboard });
+    const hasQueryParams = _.some(this.dashboard.widgets, w => !_.isEmpty(w.options.parameterMappings));
```

Comment:
TL;DR [A proper way to get widget's query parameters](https://github.com/getredash/redash/blob/master/client/app/services/widget.js#L182)

Explanation: parameter mappings were introduced recently, and on "old" setups this check will fail (widgets there does not have that field, and will not have until someone will edit it). `widget.getParametersDefs()` returns only local parameters for the widget, `dashboard.getParametersDefs()` - only dashboard-level parameters. So the proper way to know if there are any parameters - to get it from widgets query (if exists - textboxes does not have query attached).

Reply:
Fixed @kravets-levko

Reply:
Yeah, I see :+1: